### PR TITLE
Closer to cran

### PR DESCRIPTION
**Pull request by martin-ueding**
I have made a few more tweaks in order to remove more warnings. We are not there yet, but now the main thing missing is documentation of functions that I do not know.

Also I think it would not be clever for me to submit but Carsten should do that in order to become the owner of that package. We then should do the same with rzeta and paramvalf such that students can just install them all from CRAN.

### CODE REVIEW

File: NAMESPACE [hunk 1]
```diff
@@ -47,3 +47,53 @@ importFrom("stats", "acf", "approx", "complete.cases", "cov", "deltat",
...
+S3method(summary, effectivemass)
+S3method(summary, effectivemassfit)
+S3method(summary, gevp.amplitude)
+S3method(summary, hadronacf)
+S3method(summary, matrixfit)
+S3method(summary, ofit)
+S3method(summary, pionff)
+S3method(summary, pionfit)
+S3method(summary, raw_cf)
+S3method(summary, uwerr)
```
Comment (urbach) [inline NAMESPACE]: why is this needed?

File: R/raw_cf.R [hunk 2]
```diff
@@ -557,7 +557,7 @@ get_plotdata_raw_cf <- function(cf,
 #' @param reim_same Boolean, determines whether the real and imaginary parts, if both
 #'                  are to be plotted, will be plotted in the same plot.
 #' @param ... Further parameters passed to \link{plotwitherror}.
-#' @return Invisibly returns the plotdata, see \link{get_plotdata.raw_cf}.
+#' @return Invisibly returns the plotdata, see \link{get_plotdata_raw_cf}.
```
Comment (urbach) [inline R/raw_cf.R]: '.' here, but not in `plot.raw_cf` and others?

Comment (kostrzewa) [reply R/raw_cf.R]: The choice between `.` and `_` is based on whether a generic function exists or not not. Since `get_plotdata_raw_cf` is a utility function (which is probably not even exported), it makes sense for it to not use the dot notation. On the other hand, `plot.raw_cf` can be dispatched by the `plot` generic function. Is that what you meant?

Comment (kostrzewa) [reply R/raw_cf.R]: Things are not quite consistent, however, as exemplified by `uwerr`...

Comment (kostrzewa) [reply R/raw_cf.R]: And there are some generic functions missing, for which it might actually be interesting to provide prototypes.

**Comment by urbach** [comment]
do we need in your opinion to export the following functions:

- compute.plotlims
- compute.qtildesq
- compute.qtildesq.contdisp
- effmass2
- g1
- pcacfit
- read_nissa_textcf_kernel
- symmetrise.cf
- uwerr.cf

**Comment by kostrzewa** [comment]
> * read_nissa_textcf_kernel

no, that's a mistake


> * symmetrise.cf

yes

> * uwerr.cf

yes

**Comment by urbach** [comment]
so, the docu is fixed and I merged to master. We could merge this in, I think it has no consequences other than more docu!?

File: R/analysis_online.R [hunk 3]
```diff
@@ -15,6 +15,66 @@ append_pdf_filename <- function(basename, pdf_filenames){
...
+#' 
+#' @param L integer. spatial lattice extend
+#' @param T integer. temporal lattice extend
+#' @param t1 integer. initial time of fit range
+#' @param t2 integer. end time of fit range
+#' @param beta numeric. inverse squared gauge coupling
+#' @param kappa numeric. hopping parameter
+#' @param mul numeric. light twisted mass
+#' @param cg_col integer. column of CG iteration counts to use
+#' @param evals ??. ??
```
Comment (urbach) [inline R/analysis_online.R]: @kostrzewa could you provide a replacement for ??

File: R/analysis_online.R [hunk 4]
```diff
@@ -15,6 +15,66 @@ append_pdf_filename <- function(basename, pdf_filenames){
...
+#' @param oldnorm boolean. If `TRUE` use the old tmLQCD normalisation
+#' @param S numeric. `S` parameter of \link{uwerr}
+#' @param stat_skip integer. statistics to skip
+#' @param omeas.samples integer. number of stochastic sample per online measurement
+#' @param omeas.stride integer. use as strides in online measurements
+#' @param omeas.avg integer. ??
+#' @param omeas.start integer. ??
+#' @param omeas.stepsize integer. ??
+#' @param evals.start integer. ??
+#' @param evals.stepsize integer. ??
```
Comment (urbach) [inline R/analysis_online.R]: @kostrzewa also here?

File: R/raw_cf.R (hunk 2 above)
Comment (urbach) [reply R/raw_cf.R]: okay, I got it

File: NAMESPACE (hunk 1 above)
Comment (martin-ueding) [reply NAMESPACE]: When one does not attaches the namespace, the S3 method resolution does not work:

```
> plot(hadron::samplecf)
Fehler in xy.coords(x, y, xlabel, ylabel, log) : 
  'x' is a list, but does not have components 'x' and 'y'
```

With these explicit S3 methods, the above should work, but I have not tested it.

File: R/analysis_online.R (hunk 4 above)
Comment (kostrzewa) [reply R/analysis_online.R]: done

**Comment by kostrzewa** [comment]
Man, `analysis_online` is such a steaming pile of ...

**Comment by kostrzewa** [comment]
Running checks on the various additions now.

**Comment by kostrzewa** [comment]
We're very close now, just a few exported objects remaining to be documented.

**Comment by urbach** [comment]
> Man, `analysis_online` is such a steaming pile of ...

yep!

**Comment by kostrzewa** [comment]
> Man, `analysis_online` is such a steaming pile of ...
> yep!

I'd really like to modularise it and add support for multi-replica analysis, but that will probably take several days which are simply not on the horizon right now...

**Comment by urbach** [comment]
do we want to export

‘weight.cf’ and ‘weight_shift_reweight.cf’

?

**Comment by martin-ueding** [comment]
Yes please, I use them in the LüscherAnalysis package.

**Comment by urbach** [comment]
> Yes please, I use them in the LüscherAnalysis package.

I think you wrote them, so, could you maybe, if it doesn't create too
much of a problem ... write the docu?!

**Comment by urbach** [comment]
see pull request #198

**Comment by urbach** [comment]
is `devtools:load_all()` in `jackknife_cov_and_missing_values.Rmd`
really needed?

**Comment by urbach** [comment]
and given this here from R CMD check:

gevp2cf: possible error in cf_weighted(cf, weight.factor =
  gevp$cf$weight.factor, weight.cosh = gevp$cf$weight.cosh, mass1 =
  gevp$cf$mass1, mass2 = gevp$cf$mass2): unused arguments (mass1 =
  gevp$cf$mass1, mass2 = gevp$cf$mass2)

Maybe we should still merge the fix from @ueding in?

**Comment by martin-ueding** [comment]
And yes, we should merge the fix from me there, I'll see about the failing unit tests. In the current form the function is just wrong for `cosh.factor != 0.0`.

**Comment by urbach** [comment]
one thing left:

    Codoc mismatches from documentation object 'weight.cf':
    weight.cf
      Code: function(cf, energy_difference_val, energy_difference_boot,
                     cosh_factor, offset = 0, inverse = FALSE)
      Docs: function(cf, energy_difference_val, energy_difference_boot,
                     cosh_factor, offset = 0)
      Argument names in code not in docs:
        inverse

**Comment by urbach** [comment]
it was still the old docu in the repo...

**Comment by martin-ueding** [comment]
I believe that this was because the function has been changed in a different branch. I'll fix it.
